### ISSUE TITLE
SMS are randomly delivered on concurrent sending

### ISSUE THREAD

**Issue by robo-w** [n0r1sk/smsgateway#51]
Message with same content should be sent to multiple recipients:
* Javascript client iterates over recipients, calls send URL for each recipient

If the calls are executed at the same time, only the first address gets a message, all other messages are lost and are not visible in the outgoing message list.

If the calls are queued and delayed by 500 to 2000 milliseconds between each call, most of the recipients get messages, but some get duplicate messages. In the list of the outgoing messages, all recipients are visible.

**Comment by m4r10k** [comment]
Yes, you are correct. It works better if you use a delay during your send calls or you can use multiple devices to spread the text messages for example. The code is far from being perfect at this point. Are you using multiple WIS instances or are you using a singular WIS? The WIS is holding the database and is responsible for the operations against it.

Nowadays it would be better to use something like a distributed ETCD with appropriate concurrency locking instead of self managed multiple Sqlite db's. In the past we thought about it to use something like https://github.com/rqlite/rqlite alternatively.

So yes, there would be a lot of work to resolve this issue sustainably.
